mgmtd/mgmt_db.h

```diff
@@ -0,0 +1,432 @@
+/*
+ * MGMTD Databases
+ * Copyright (C) 2021  Vmware, Inc.
+ *		       Pushpasis Sarkar <spushpasis@vmware.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef _FRR_MGMTD_DB_H_
+#define _FRR_MGMTD_DB_H_
+
+#include "mgmtd/mgmt_defines.h"
+#include "mgmtd/mgmt_bcknd_adapter.h"
+#include "mgmtd/mgmt_frntnd_adapter.h"
+
+#define MGMTD_MAX_NUM_DBNODES_PER_BATCH 128
+
+#define MGMTD_DB_NAME_MAX_LEN 32
+#define MGMTD_DB_NAME_NONE "none"
+#define MGMTD_DB_NAME_RUNNING "running"
+#define MGMTD_DB_NAME_CANDIDATE "candidate"
+#define MGMTD_DB_NAME_OPERATIONAL "operational"
+
+#define MGMTD_STARTUP_DB_FILE_PATH "/etc/frr/frr_startup.json"
+
+#define FOREACH_MGMTD_DB_ID(id)                                                \
+	for ((id) = MGMTD_DB_NONE; (id) < MGMTD_DB_MAX_ID; (id)++)
+
+#define MGMTD_MAX_COMMIT_LIST 10
+#define MGMTD_MD5_HASH_LEN 16
+#define MGMTD_MD5_HASH_STR_HEX_LEN 33
+
+#define MGMTD_MAX_COMMIT_FILE_PATH_LEN 55
+#define MGMTD_COMMIT_FILE_PATH "/etc/frr/commit-%s.json"
+#define MGMTD_COMMIT_INDEX_FILE_NAME "/etc/frr/commit-index.dat"
```

Comment:
`/etc` is not a path to create database-type files under.

Reply:
What would you suggest then? Whatever you suggest the path should be writable from FRR daemons. I think we couple of other paths but they were not writable from mgmtd.

Reply:
`/var/lib/frr/<namespace>/` is the "correct" place for this according to [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard)

> State information. Persistent data modified by programs as they run (e.g., databases, packaging system metadata, etc.).

Reply:
Also this must respect `--localstatedir` from `./configure`

Reply:
Hmm. Good point let us see how we can make it happen.